src/Stratis.Bitcoin.Features.Miner/Staking/PosMinting.cs

```diff
@@ -1115,27 +1143,39 @@ public Models.GetStakingInfoModel GetGetStakingInfoModel()
+        /// but aiming not to exceed a total amount of coins bigger than (500+1)*3. S
         /// </summary>
-        /// <param name="utxoCount">Number of non-empty UTXOs in the wallet.</param>
-        /// <param name="chainTip">The chain tip</param>
+        /// <param name="stakedUtxosCount">Number of non-empty UTXOs currently at stake.</param>
+        /// <param name="amountStaked">Total amount currently at stake</param>
+        /// <param name="coinValue">Value of the coin we are considering to split</param>
+        /// <param name="splitFactor">number of time we want to split the coin</param>
+        /// <param name="chainHeight">Current height of the chain</param>
         /// <returns><c>true</c> if the coinstake should be split, <c>false</c> otherwise.</returns>
-        /// <remarks>The coinstake is split if the number of non-empty UTXOs we have in the wallet
-        /// is under the given threshold.</remarks>
+        /// <remarks>The coinstake is split if the number of non-empty UTXOs we have in the wallet is under the given threshold, or if amount is above a certain value.</remarks>
         /// <seealso cref="CoinstakeSplitLimitMultiplier"/>
-        private bool GetSplitStake(int utxoCount, ChainedHeader chainTip)
+        internal bool ShouldSplitStake(int stakedUtxosCount, long amountStaked, long coinValue, int splitFactor, int chainHeight)
         {
-            this.logger.LogTrace("({0}:{1})", nameof(utxoCount), utxoCount);
-
-            long maturityLimit = this.network.Consensus.CoinbaseMaturity;
-            long coinAgeLimit = ((PosConsensusOptions)this.network.Consensus.Options).GetStakeMinConfirmations(chainTip.Height + 1, this.network);
-            long requiredCoinAgeForStaking = Math.Max(maturityLimit, coinAgeLimit);
+            this.logger.LogTrace("({0}:{1},{2}:{3},{4}:{5},{6}:{7})", nameof(stakedUtxosCount), stakedUtxosCount, nameof(amountStaked), amountStaked, 
+                nameof(coinValue), coinValue, nameof(splitFactor), splitFactor);
+            
+            long coinAgeLimit = ((PosConsensusOptions)this.network.Consensus.Options).GetStakeMinConfirmations(chainHeight + 1, this.network);
+            long requiredCoinAgeForStaking = Math.Max(this.network.Consensus.CoinbaseMaturity, coinAgeLimit);
             this.logger.LogTrace("Required coin age for staking is {0}.", requiredCoinAgeForStaking);
 
-            bool res = utxoCount < (requiredCoinAgeForStaking + 1) * CoinstakeSplitLimitMultiplier;
+            long targetCoinDistributionSize = (requiredCoinAgeForStaking + 1) * CoinstakeSplitLimitMultiplier;
 
-            this.logger.LogTrace("(-):{0}", res);
-            return res;
+            bool coinAboveMaxValue = coinValue > splitFactor * 100 * Money.COIN;
+            bool utxosCanStillBeSplitted = stakedUtxosCount < targetCoinDistributionSize;
+            bool coinAboveTargetAverage = coinValue > amountStaked / targetCoinDistributionSize;
+
```

Comment:
Here I am not sure if we want this or : coinAboveMaxValue || (utxosCanStillBeSplitted && coinAboveTargetAverage);
I am thinking that in very very rich wallets we would probably end up with too many utxos for no real benefit. I might have misunderstood the issue, but I thought once we have something like 1503 roughly even UTXOs for staking we might not care if they are being "attacked".

Reply:
I think utxosCanStillBeSplitted  shouldn't be there at all

Reply:
It is because not only through attack you can get small coins and those small coins you get normally could be higher than the limit for staking